static/js/components/material/Filefield.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+import React from "react";
+
+export default class Filefield extends React.Component {
+  fileInput: null;
+  props: {
+    label?: string,
+    accept?: string,
+    className?: string,
+  };
+
+  handleClick = () => {
+    if (this.fileInput) {
+      const uploadBtn = this.fileInput;
+      uploadBtn.value = null;
+      uploadBtn.click();
+    }
+  };
+
+  render() {
+    const { label, accept, className, ...otherProps } = this.props;
+
+    let acceptedTypes = accept ? accept : "*";
+
+    return <a onClick={this.handleClick} href="#"
+      className={className ? `${className} button-link upload-link` : "button-link upload-link"}>
+      <input type="file" className="hidden upload-input"
+        accept={acceptedTypes} {...otherProps} ref={(input) => { this.fileInput = input; }}/>
+      {label}
+    </a>;
```

Comment:
Mentioned this elsewhere but I think our preference for components/elements with a lot of attributes is to put each attribute indented on a new line and to put a spread operator (`{...otherProps}`) at the end

Reply:
OK